src/Sets/Star.jl

```diff
@@ -0,0 +1,211 @@
+export Star,
+       center,
+       basis,
+       predicate
+
+"""
+    Star{N, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}, PT<:AbstractPolyhedron{N}} <: AbstractStar{N}
+
+Type that represents a generalized star set where the predicate is polyhedral, i.e.
+
+```math
+X = \\{x ∈ \\mathbb{R}^n : x = x₀ + \\sum_{i=1}^m α_i v_i,~~\\textrm{s.t. } P(α) = ⊤ \\},
+```
+where ``x₀ ∈ \\mathbb{R}^n`` is the center, the ``m`` vectors ``v₁, …, vₘ`` form
+the basis of the star set, and ``P : α ∈ \\mathbb{R}^m → \\{⊤, ⊥\\}`` where
+``P(α) = ⊤`` if and only if ``Aα ≤ b`` is the polyhedral predicate.
+
+### Fields
+
+- `c` -- vector that represents the center
+- `V` -- matrix where each column corresponds to a basis vector
+- `P` -- polyhedral set that represents the predicate
+
+### Notes
+
+The predicate function is implemented as a conjunction of linear constraints, i.e.
+a subtype of `AbstractPolyhedron`. By a slight abuse of notation, the *predicate*
+is also used to denote the subset of ``\\mathbb{R}^n`` such that ``P(α) = ⊤`` holds.
+
+The ``m`` basis vectors (each one ``n``-dimensional) are stored as the columns
+of an ``n × m`` matrix.
+
+### Examples
+
+This example is drawn from Example 1 in [2]. Consider the two-dimensional plane
+``\\mathbb{R}^2``. Let
+
+```jldoctest star_constructor
+julia> V = [[1.0, 0.0], [0.0, 1.0]]
+2-element Array{Array{Float64,1},1}:
+ [1.0, 0.0]
+ [0.0, 1.0]
+```
+be the bais vectors and take
+
+```jldoctest star_constructor
+julia> c = [3.0, 3.0]
+2-element Array{Float64,1}:
+ 3.0
+ 3.0
+```
+as the center of the star set. Let the predicate be the infinity-norm ball of radius 1,
+
+```jldoctest star_constructor
+julia> P = BallInf(zeros(2), 1.0)
+BallInf{Float64,Array{Float64,1}}([0.0, 0.0], 1.0)
+```
+Finally, the star set ``X = ⟨c, V, P⟩`` defines the set:
+
+```jldoctest star_constructor
+julia> S = Star(c, V, P)
+Star{Float64,Array{Float64,1},Array{Float64,2},BallInf{Float64,Array{Float64,1}}}([3.0, 3.0], [1.0 0.0; 0.0 1.0], BallInf{Float64,Array{Float64,1}}([0.0,
+0.0], 1.0))
+```
+
+We can use getter functions for each component field:
+
+```jldoctest star_constructor
+julia> center(S)
+2-element Array{Float64,1}:
+ 3.0
+ 3.0
+
+ julia> basis(S)
+ 2×2 Array{Float64,2}:
+  1.0  0.0
+  0.0  1.0
+
+julia> predicate(S)
+BallInf{Float64,Array{Float64,1}}([0.0, 0.0], 1.0)
+```
+In this case, we know calculating by hand that the generalized star ``X``
+defines the rectangular set:
+
+```math
+ Θ = \\{(x, y) ∈ \\mathbb{R}^2 : (2 ≤ x ≤ 4) ∧ (2 ≤ y ≤ 4) \\}
+```
+We can check that assertion:
+
+```julia
+julia> Θ = Hyperrectangle(low=[2, 2.], high=[4, 4.])
+Hyperrectangle{Float64,Array{Float64,1},Array{Float64,1}}([3.0, 3.0], [1.0, 1.0])
+
+julia> isequivalent(X, Θ) # FIXME
```

Comment:
Something missing here? Also, `Θ` looks like a 0 in the code. Maybe just use `Y`?

Reply:
this requires some prior work in `isequivalent` and we need to add an inclusion check for stars. but i can make a new issue out of this  comment, that's probably a better approach.